bitcoin_client_rs/examples/ledger_hwi/src/main.rs

```diff
@@ -103,6 +129,30 @@ async fn register_wallet<T: Transport>(
     Ok(())
 }
 
+async fn sign<T: Transport>(
+    client: &BitcoinClient<T>,
+    psbt: &str,
+    name: &str,
+    policy: &str,
+    hmac: Option<&str>,
+) -> Result<(), Box<dyn Error>> {
+    let psbt: Psbt = deserialize(&base64::decode(&psbt).unwrap()).unwrap();
+    let (descriptor_template, keys) = extract_keys_and_template(policy)?;
+    let wallet = WalletPolicy::new(name.to_string(), Version::V2, descriptor_template, keys);
+    let hmac = hmac.map(|s| {
+        let mut h = [b'\0'; 32];
+        h.copy_from_slice(&Vec::from_hex(&s).unwrap());
```

Comment:
Same, `?` for `unwrap`?

There are a few more cases of `unwrap` when the function is returning a `Result`, not commenting further.